Sources/Core2/SubscriberConfigurator.cs

```diff
@@ -151,17 +165,50 @@ public ISubscriberConfigurator UseTrace(ITrace trace)
 
         public ISubscriberConfigurator UseDurableQueue(string queueName)
         {
+            _durable = true;
+
             _queueName = queueName;
 
+            _autoCreate = false;
+
             return this;
         }
 
         public ISubscriberConfigurator UseDurableQueue(string queueName, bool createBindings)
         {
+            _durable = true;
+
             _queueName = queueName;
 
             _createBindings = createBindings;
 
+            _autoCreate = false;
+
+            return this;
+        }
+
+        public ISubscriberConfigurator UseDurableQueue(string queueName, bool createBindings, bool autoCreate)
+        {
+            _durable = true;
+
+            _queueName = queueName;
+
+            _createBindings = createBindings;
+
+            _autoCreate = autoCreate;
+
+            return this;
+        }
+
+        public ISubscriberConfigurator UseNonDurableNamedQueue(string queueNameSuffix)
+        {
+            _durable = false;
+
+            _autoCreate = true;
+
+            _queueName = "serviceBus.gen-" + System.Environment.MachineName

```

Comment:
Since the queue created in this case is non-durable, using a constant name could lead to bugs e.g. providing an existing queue's name. I thought that giving a volatile  prefix to the requested queue name could prevent misusage. I could easily move this logic to the calling code if you think this is best.